internal/analysis/common.go

```diff
@@ -154,18 +154,16 @@ func collectAnalysisStats(fullSubmissionIDs []string, totalRunTime int64, initia
 	now := timestamp.Now()
 
 	for courseID, assignmentID := range seenIdentifiers {
-		metric := stats.CourseMetric{
-			BaseMetric: stats.BaseMetric{
-				Timestamp: now,
-				Attributes: map[string]any{
-					stats.ATTRIBUTE_KEY_ANALYSIS: analysisType,
-				},
+		metric := stats.BaseMetric{
+			Timestamp: now,
+			Attributes: map[string]any{
+				stats.ATTRIBUTE_KEY_ANALYSIS: analysisType,
+				stats.TYPE_KEY:               stats.CourseMetricTypeCodeAnalysisTime,
+				stats.COURSE_ID_KEY:          courseID,
+				stats.ASSIGNMENT_ID_KEY:      assignmentID,
+				stats.USER_EMAIL_KEY:         initiatorEmail,
+				stats.VALUE_KEY:              uint64(totalRunTime),
```

Comment:
You BaseMetric needs work.

Why would `stats.TYPE_KEY` and `stats.VALUE_KEY` be entries in the map?
Wouldn't these be in every matric, and therefore be first-class fields?